cf/samples/javascript-samples/geotab/geotab-to-sap-cloud.js

```diff
@@ -0,0 +1,768 @@
+                // console.debug('First record:', JSON.stringify(result.data[0], null, 4));
+                resolve(result);
+            }
+        });
+    });
+}
+
+/**
+ * Look up the device by its ID in the Geotab API.
+ *
+ * @param {string} geotabDeviceId the Geotab device ID
+ */
+async function lookupDevice(geotabDeviceId) {
+    return new Promise(function (resolve, reject) {
+        geotab.call('Get', {
+            typeName: 'Device',
+            search: {
+                id: geotabDeviceId
+            }
+        }, function (error, result) {
+            if (error) {
+                console.error('Got error:', error);
+                reject(Error(error));
+            } else {
+                // console.debug('Got result:', result);
+                if (result.length > 0) {
+                    resolve(result[0]);
+                } else {
+                    reject(Error('Device not found!'));
+                }
+            }
+        });
+    });
+}
+
+/**
+ * Look up the Geotab Diagnostic by its ID.
+ *
+ * @param {string} diagnosticId the Geotab Diagnostic ID
+ */
+async function lookupDiagnostic(diagnosticId) {
+    return new Promise(function (resolve, reject) {
+        geotab.call('Get', {
+            typeName: 'Diagnostic',
+            search: {
+                id: diagnosticId
+            }
+        }, function (error, result) {
+            if (error) {
+                console.error('Got error:', error);
+                reject(Error(error));
+            } else {
+                // console.debug('Got result:', result);
+                if (result.length > 0) {
+                    resolve(result[0]);
+                } else {
+                    reject(Error('Diagnostic not found!'));
+                }
+            }
+        });
+    });
+}
+
+/**
+ * Post an array of LogRecord objects to SAP IoT Service.
+ * @param {!Object[]} records  the records to post
+ */
+async function post_LogRecords(records) {
+    const measurements = {}; // Key: vehicle identification number (VIN), value: an array of measurements
+    for (const record of records) {
+        const geotabDeviceId = record.device.id;
+        // The device alternate IDs used by SAP IoT Services are generally supposed to have some
+        // real-world meaning, so that they can be derived from real-world data. We use VIN numbers
+        // to identify vehicles.
+        let vin = geotabDeviceCache[geotabDeviceId];
+        if (!vin) {
+            // Cache lookup failed.
+            try {
+                const device = await lookupDevice(geotabDeviceId);
+                // console.debug('Got device:', device);
+                vin = device.vehicleIdentificationNumber;
+                if (!vin) {
+                    console.error(`Geotab device with ID ${geotabDeviceId} has no vehicleIdentificationNumber, skipped!`);
+                   continue;
```

Comment:
check if it's possible to fix it: https://rules.sonarsource.com/javascript/RSPEC-909?search=continue